File: rrouter-core/src/main/java/org/sheedon/rrouter/Converter.java
package org.sheedon.rrouter;

import org.sheedon.rrouter.core.support.IRspModel;

/**
 * 数据转换
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/2/15 7:35 下午
 */
public interface Converter<T, F> {

    F convert(T t);


    class Factory {

        Converter<IRspModel<?>, Boolean> createCheckConverter() {
            return null;
        }

    }
}

File: rrouter-core/src/main/java/org/sheedon/rrouter/core/Converter.kt
/*
 * Copyright (C) 2022 Sheedon.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.sheedon.rrouter.core

/**
 * 数据转换
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/2/15 7:35 下午
 */
interface Converter<T, F> {
    fun convert(t: T): F
    open class Factory {
        open fun createCheckConverter(): Converter<*, IRspModel<*>?>? {
            return null
        }
    }
}